akka-docs/src/test/scala/docs/stream/operators/source/Restart.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.stream.operators.source
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.KillSwitches
+import akka.stream.UniqueKillSwitch
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.RestartSource
+import akka.stream.scaladsl.Sink
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+// #imports
+import akka.stream.scaladsl.Source
+// #imports
+
+object Restart extends App {
+  implicit val system: ActorSystem = ActorSystem()
+
+  onRestartWitFailureKillSwitch()
+
+  case class CantConnectToDatabase(msg: String) extends RuntimeException(msg) with NoStackTrace
+
+  def onRestartWithBackoffInnerFailure(): Unit = {
+    //#restart-failure-inner-failure
+    // could throw if for example it used a database connection to get rows
+    val flakySource: Source[Int, NotUsed] =
+      Source(List(() => 1, () => 2, () => 3, () => throw CantConnectToDatabase("darn"))).map(_())
```

Comment:
This indirection of using `map` to create the factory could be confusing.

Wouldn't it be more clear to define it as `val flakySource: () => Source[Int, NotUsed]

Reply:
and same in other places, including java sample

Reply:
Or Source[() => Int, NotUsed] ?